nixos/modules/tasks/network-interfaces.nix

```
@@ -68,13 +68,18 @@ let
   # We must escape interfaces due to the systemd interpretation
   subsystemDevice = interface: "sys-subsystem-net-devices-${escapeSystemdPath interface}.device";
 
+  addrTypeDynamic =
+    v:
+    assert v == 4 || v == 6;
+    if v == 4 then lib.types.address.ipv4 else lib.types.address.ipv6;
```

Might be better to do this:
```suggestion
  addrTypeDynamic =
    v:
    if v == 4 then lib.types.address.ipv4 else if v == 6 then lib.types.address.ipv6 else lib.throw "...";
```